src/shared/types/checkReactTypeSpec.js

```diff
@@ -34,101 +28,44 @@ if (
   ReactComponentTreeHook = require('ReactComponentTreeHook');
 }
 
-var loggedTypeFailures = {};
-
-/**
- * Assert that the values match with the type specs.
- * Error messages are memorized and will only be shown once.
- *
- * @param {object} typeSpecs Map of name to a ReactPropType
- * @param {object} values Runtime values that need to be type-checked
- * @param {string} location e.g. "prop", "context", "child context"
- * @param {string} componentName Name of the component for error messages.
- * @param {?object} element The React element that is being type-checked
- * @param {?number} workInProgressOrDebugID The React component instance that is being type-checked
- * @private
- */
 function checkReactTypeSpec(
   typeSpecs,
   values,
-  location: ReactPropTypeLocations,
+  location: string,
   componentName,
   element,
   // It is only safe to pass fiber if it is the work-in-progress version, and
   // only during reconciliation (begin and complete phase).
   workInProgressOrDebugID,
+  warnOnRepeat
 ) {
-  for (var typeSpecName in typeSpecs) {
-    if (typeSpecs.hasOwnProperty(typeSpecName)) {
-      var error;
-      // Prop type validation may throw. In case they do, we don't want to
-      // fail the render phase where it didn't fail before. So we log it.
-      // After these have been cleaned up, we'll let them throw.
-      try {
-        // This is intentionally an invariant that gets caught. It's the same
-        // behavior as without this statement except with a better message.
-        invariant(
-          typeof typeSpecs[typeSpecName] === 'function',
-          '%s: %s type `%s` is invalid; it must be a function, usually from ' +
-          'React.PropTypes.',
-          componentName || 'React class',
-          ReactPropTypeLocationNames[location],
-          typeSpecName
-        );
-        error = typeSpecs[typeSpecName](values, typeSpecName, componentName, location, null, ReactPropTypesSecret);
-      } catch (ex) {
-        error = ex;
+  function formatMessage(message) {
```

Comment:
I think this could be designed with global context. In Fiber we went with an approach that picked up the current context from a global mutable.

https://github.com/facebook/react/blob/e36b38c1cad22f7dff557736cf4b0280106e937e/src/renderers/dom/fiber/wrappers/ReactDOMFiberTextarea.js#L72

That way we don't have to pass it along as an additional argument everywhere.

If it did it that way this function could just be a global instead of creating a closure.

It seems natural for things like ReactElementValidator anyway since we read the owner from `ReactCurrentOwner` anyway. We just pass it in a very round about way.